log/DESIGN.md

```diff
@@ -0,0 +1,440 @@
+The interface may have methods added without a package major version bump.
+The embedds `embedded.LoggerProvider` to help inform an API implementation
+author about this non-standard API evolution.
+This approach is already used in Trace API and Metrics API.
+
+#### LoggerProvider.Logger
+
+The `Logger` method implements the [`Get a Logger` operation](https://opentelemetry.io/docs/specs/otel/logs/bridge-api/#get-a-logger).
+
+The required `name` parameter is accepted as a `string` method argument.
+
+The following options are defined to support optional parameters:
+
+```go
+func WithInstrumentationVersion(version string) LoggerOption
+func WithInstrumentationAttributes(attr ...attribute.KeyValue) LoggerOption
+func WithSchemaURL(schemaURL string) LoggerOption
+```
+
+Implementation requirements:
+
+- The [specification requires](https://opentelemetry.io/docs/specs/otel/logs/bridge-api/#concurrency-requirements)
+  the method to be safe to be called concurrently.
+
+- The method should use some default name if the passed name is empty
+  in order to meet the [specification's SDK requirement](https://opentelemetry.io/docs/specs/otel/logs/sdk/#logger-creation)
+  to return a working logger when an invalid name is passed
+  as well as to resemble the behavior of getting tracers and meters.
+
+`Logger` can be extended by adding new `LoggerOption` options
+and adding new exported fields to the `LoggerConfig` struct.
+This design is already used in Trace API for getting tracers
+and in Metrics API for getting meters.
+
+Rejected alternative:
+
+- [Passing struct as parameter to LoggerProvider.Logger](#passing-struct-as-parameter-to-loggerproviderlogger).
+
+### Logger
+
+The [`Logger` abstraction](https://opentelemetry.io/docs/specs/otel/logs/bridge-api/#logger)
+is defined as an interface:
+
+```go
+type Logger interface {
+	embedded.Logger
+	Emit(ctx context.Context, record Record)
+}
+```
+
+The specification may add new operations to `Logger`.
+The interface may have methods added without a package major version bump.
+The embedds `embedded.Logger` to help inform an API implementation
+author about this non-standard API evolution.
+This approach is already used in Trace API and Metrics API.
+
+### Logger.Emit
+
+The `Emit` method implements the [`Emit a LogRecord` operation](https://opentelemetry.io/docs/specs/otel/logs/bridge-api/#emit-a-logrecord).
+
+[`Context` associated with the `LogRecord`](https://opentelemetry.io/docs/specs/otel/context/)
+is accepted as a `context.Context` method argument.
+
+Calls to `Emit` are supposed to be on the hot path.
+Therefore, in order to reduce the number of heap allocations,
+the [`LogRecord` abstraction](https://opentelemetry.io/docs/specs/otel/logs/bridge-api/#emit-a-logrecord),
+is defined as a `struct`:
+
+```go
+type Record struct {
+	Timestamp         time.Time
+	ObservedTimestamp time.Time
+	Severity          Severity
+	SeverityText      string
```

Comment:
Do you find it acceptable to keep the `SeverityNumber` constant as e.g. `SeverityTrace2` and NOT `SeverityNumberTrace2` (I would find it unnecessarily verbose).

Reply:
As far as I can tell these are actually independent in the API.  While the data-model says you can use the short name for the Text if it's empty and the Number is populated, the API only says that you can accept both at the same time.  

Because we will probably use the short name somewhere, I would suggest that we use `stringer` to make the short names.

Reply:
> As far as I can tell these are actually independent in the API. While the data-model says you can use the short name for the Text if it's empty and the Number is populated, the API only says that you can accept both at the same time.

That is indeed important in scope of https://github.com/open-telemetry/opentelemetry-go/pull/4809#issuecomment-1887875986 as it would be then better to have them separated 👍 

> Because we will probably use the short name somewhere, I would suggest that we use stringer to make the short names.

I think this should live in the SDK as this is defined in the Data Model and it is only a recommendation in [Displaying Severity](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/logs/data-model.md#displaying-severity).

@MrAlias @MadVikingGod Are you OK with doing nothing and just resolving this thread?

Reply:
If the plan is to change to a getter/setter design, I'm fine resolving this and opening a new thread if the issue relates to the redesign.

Reply:
Resolving per https://github.com/open-telemetry/opentelemetry-go/pull/4809/commits/35ff8f784bd50d6efd5270bb6186d1d6b4953a69